rpm.go

```diff
@@ -58,42 +61,56 @@ type RPMFile struct {
 // RPM holds the state of a particular rpm file. Please use NewRPM to instantiate it.
 type RPM struct {
 	RPMMetaData
-	di          *dirIndex
-	payload     *bytes.Buffer
-	payloadSize uint
-	cpio        *cpio.Writer
-	basenames   []string
-	dirindexes  []uint32
-	filesizes   []uint32
-	filemodes   []uint16
-	fileowners  []string
-	filegroups  []string
-	filemtimes  []uint32
-	filedigests []string
-	filelinktos []string
-	closed      bool
-	gzPayload   *gzip.Writer
-	files       map[string]RPMFile
-	prein       string
-	postin      string
-	preun       string
-	postun      string
+	di                *dirIndex
+	payload           *bytes.Buffer
+	payloadSize       uint
+	cpio              *cpio.Writer
+	basenames         []string
+	dirindexes        []uint32
+	filesizes         []uint32
+	filemodes         []uint16
+	fileowners        []string
+	filegroups        []string
+	filemtimes        []uint32
+	filedigests       []string
+	filelinktos       []string
+	closed            bool
+	compressedPayload io.WriteCloser
+	files             map[string]RPMFile
+	prein             string
+	postin            string
+	preun             string
+	postun            string
 }
 
 // NewRPM creates and returns a new RPM struct.
-func NewRPM(m RPMMetaData) (*RPM, error) {
+func NewRPM(m RPMMetaData) (rpm *RPM, err error) {
+	if len(m.Compressor) == 0 {
```

Comment:
Please don't use len(string), instead compare directly to "".

Actually, come to think of it, it would be even nicer to add it as a case in your switch:
```
case "gzip", "":
  m.Compressor = "gzip"
  z, err = gzip....